### DISCUSSION TITLE
HA or Domoticz MQTT support

### DISCUSSION THREAD

**Discussion by MatchoLM** [softwarecrash/EPEver2MQTT#15] [Ideas]
Hi, is it possible to customize MQTT format to Homeassistant or domoticz ?

**Comment by softwarecrash** [comment]
hello,
no thats not possible.
for HA intregration take a look at 
https://github.com/JoergOstertag/DALY-BMS-to-MQTT/tree/main/docs/iot-integration

its written for daly2mqtt but epever is verry similar

**Comment by Ivanodr** [comment]
Adding part of my code, new entities will be created after integration into HA


- state_topic: "EPEVER1/EP_1/LiveData/CHARGER_MODE"
       unique_id: 'sensor.EPEVER1_charger_mode'
       name: "EPEVER1 Mod nabíjačky"
       icon: mdi:current-dc

     - state_topic: "EPEVER1/EP_1/StatsData/GEN_TOT"
       unique_id: 'sensor.EPEVER1_GEN_TOT'
       name: "EPEVER1 Výroba Totál"
       icon: mdi:solar-power
       unit_of_measurement: "kW"
       device_class: "power"

     - state_topic: "EPEVER1/EP_1/StatsData/GEN_YEAR"
       unique_id: 'sensor.EPEVER1_GEN_YEAR'
       name: "EPEVER1 Výroba Rok"
       icon: mdi:solar-power
       unit_of_measurement: "kW"
       device_class: "power"

     - state_topic: "EPEVER1/EP_1/StatsData/GEN_MON"
       unique_id: 'sensor.EPEVER1_GEN_MON'
       name: "EPEVER1 Výroba Mesiac"
       icon: mdi:solar-power
       unit_of_measurement: "kW"
       device_class: "power"

     - state_topic: "EPEVER1/EP_1/StatsData/GEN_DAY"
       unique_id: 'sensor.EPEVER1_GEN_DAY'
       name: "EPEVER1 Výroba Deň"
       icon: mdi:solar-power
       unit_of_measurement: "kW"
       device_class: "power"

     - state_topic: "EPEVER1/EP_1/DeviceData/BOOST_VOLTS"
       unique_id: 'sensor.EPEVER1_BOOST_VOLTS'
       name: "EPEVER1 Nastavené BOOST Napätie"
       unit_of_measurement: "V"
       icon: mdi:battery-charging-high
       device_class: "voltage"

     - state_topic: "EPEVER1/EP_1/DeviceData/FLOAT_VOLTS"
       unique_id: 'sensor.EPEVER1_FLOAT_VOLTS'
       name: "EPEVER1 Nastavené Float Napätie"
       unit_of_measurement: "V"
       icon: mdi:battery-charging-medium
       device_class: "voltage"

     - state_topic: "EPEVER1/EP_1/LiveData/DEVICE_TEMP"
       unique_id: 'sensor.EPEVER1_DEVICE_TEMP'
       name: "EPEVER1 Teplota"
       unit_of_measurement: "°C"
       icon: mdi:thermometer

     - state_topic: "EPEVER1/EP_1/LiveData/SOLAR_V"
       unique_id: 'sensor.EPEVER1_SOLAR_V'
       name: "EPEVER1 Solár V"
       unit_of_measurement: "V"
       icon: mdi:solar-power
       device_class: "voltage"

     - state_topic: "EPEVER1/EP_1/LiveData/SOLAR_A"
       unique_id: 'sensor.EPEVER1_SOLAR_A'
       name: "EPEVER1 Solár A"
       unit_of_measurement: "A"
       icon: mdi:solar-power
       device_class: "current"

     - state_topic: "EPEVER1/EP_1/LiveData/SOLAR_W"
       unique_id: 'sensor.EPEVER1_SOLAR_W'
       name: "EPEVER1 Solár W"
       unit_of_measurement: "W"
       icon: mdi:solar-power
       device_class: "Power"

     - state_topic: "EPEVER1/EP_1/LiveData/BATT_V"
       unique_id: 'sensor.EPEVER1_BATT_V'
       name: "EPEVER1 Baterka V"
       unit_of_measurement: "V"
       icon: mdi:car-battery
       device_class: "voltage"

     - state_topic: "EPEVER1/EP_1/LiveData/BATT_A"
       unique_id: 'sensor.EPEVER1_BATT_A'
       name: "EPEVER1 Baterka A"
       unit_of_measurement: "A"
       icon: mdi:car-battery
       device_class: "current"

     - state_topic: "EPEVER1/EP_1/LiveData/BATT_W"
       unique_id: 'sensor.EPEVER1_BATT_W'
       name: "EPEVER1 Baterka W"
       unit_of_measurement: "W"
       icon: mdi:car-battery
       device_class: "Power"

     - state_topic: "EPEVER1/EP_1/LiveData/BATT_TEMP"
       unique_id: 'sensor.EPEVER1_BATT_TEMP'
       name: "EPEVER1 Baterka"
       unit_of_measurement: "°C"
       icon: mdi:thermometer
